### ISSUE TITLE
Nextcloud mail 4.0.7 generates invalid smime signature

### ISSUE THREAD

**Issue by Sergeij2000** [nextcloud/mail#10440]
### Steps to reproduce

1. Setup mail account and install a SMIME-certificate
2. Send two mails to yourself, select signature for the first and encryption for the second
3. Check mails in different mail clients/apps

### Expected behavior

The signature should be valid.

### Actual behavior

The signature shows a faulty hash value

![Image](https://github.com/user-attachments/assets/0989408f-a284-4680-aa7f-b7e408e99f9e)

Encryptions works

![Image](https://github.com/user-attachments/assets/7878b609-7cb2-428f-b362-73d476a61a79)

### Mail app version

4.0.7

### Nextcloud version

30.0.2snap2

### Mailserver or service

Hetzner mail.your-server,de

### Operating system

Ubuntu 24.04.1 server

### PHP engine version

PHP 8.2

### Web server

Apache (supported)

### Database

MySQL

**Comment by kesselb** [comment]
Hi @Sergeij2000, 

we did a couple of fixes to our smime code recently. 

Are you still seeing that issue?

**Comment by Sergeij2000** [comment]
No, everything works fine since several versions, so i thing the issue can be closed from my side.

Thanks and best regards,

Jo

**Comment by nextcloud-command** [comment]
Hi there! It looks like this issue hasn’t seen much activity for a while and might need a bit more information to move forward. If we don’t hear back soon, we may close it to keep things organized. Thanks again for your contributions – we really appreciate it!
